### PR TITLE
Mafia: current month's lb is not visible

### PR DESCRIPTION
**Pull request by whalemer**

### CODE REVIEW

**Comment by ZestOfLife** [comment]
Probably we should have points not visible when given as they could be counted by other users. Also this should be done for MVP Ladder too?

**Comment by whalemer** [comment]
It is done for the mvpladder. People can still always get the winners from reading the game, I don't think that not showing the points being given will help much.

File: chat-plugins/mafia.js [hunk 1]
```diff
@@ -907,17 +907,27 @@ exports.pages = {
...
+		if (!logs[ladder.section][month] || !Object.keys(logs[ladder.section][month]).length) {
+			buf += `${ladder.title} for ${date.toLocaleString("en-us", {month: 'long'})} ${date.getFullYear()} not found.</div>`;
 			return buf;
 		}
 		const keys = Object.keys(logs[ladder.section][month]).sort((a, b) => {
 			a = logs[ladder.section][month][a];
 			b = logs[ladder.section][month][b];
 			return b - a;
 		});
+		if (!(user.can('mute', null, Rooms('mafia')) || month !== (new Date()).toLocaleString("en-us", {month: "numeric", year: "numeric"}))) {
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: This should only run if the user cannot mute in mafia AND the month the user is viewing is the current month

so use && and month ===

File: chat-plugins/mafia.js [hunk 2]
```diff
@@ -907,17 +907,27 @@ exports.pages = {
...
+			const userIndex = keys.indexOf(user.userid);
+			if (userIndex !== -1) {
+				buf += `<table style="margin-left: auto; margin-right: auto"><tbody><tr><th colspan="2"><h2 style="margin: 5px auto">Mafia ${ladder.title} for ${date.toLocaleString("en-us", {month: 'long'})} ${date.getFullYear()}</h1></th></tr>`;
+				buf += `<tr><th>User</th><th>Position</th></tr>`;
+				buf += `<tr><td>${user.userid}</td><td>${userIndex + 1}</td></tr></table>`;
+			}
+			buf += `You cannot see the position of other people on the leaderboard until the month is over.</div>`;
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: Put this in the table as well, colspan="2" will work.

**Comment by HoeenCoder** [comment]
Oh and this should not affect host logs or play logs

**Comment by whalemer** [comment]
It shouldn't anyway, since you necessarily need staff to see logs.

File: chat-plugins/mafia.js (hunk 2 above)
Comment (whalemer) [reply chat-plugins/mafia.js]: The table doesn't necessarily exist then (if the user doesnt have any points), and IMO having an empty table is worse than having that text out of table.

**Comment by ZestOfLife** [comment]
Given that people can just count their games but they can’t count all the games but __bots__ can.

**Comment by whalemer** [comment]
fair enough, updated

File: chat-plugins/mafia.js [hunk 3]
```diff
@@ -1322,6 +1324,14 @@ exports.commands = {
 			this.sendReplyBox(`${showOrl ? `Original Rolelist: ` : `Rolelist: `}${roleString}`);
 		},
 
+		spectate: 'view',
+		view: function (target, room, user, connection) {
```
Comment (HoeenCoder) [inline chat-plugins/mafia.js]: `!mafia` covers this already.

**Pull request merged**